app/com/gu/floodgate/contentsource/ContentSourceApi.scala

```diff
@@ -66,12 +70,27 @@ class ContentSourceApi(contentSourceService: ContentSourceService, reindexServic
     }
   }
 
+  def cancelReindex(id: String) = Action.async { implicit request =>
+    reindexService.cancelReindex(id) map { happyOrError =>
+      happyOrError match {
+        case Good(_) => Ok
+        case Bad(error) => BadRequest(Json.toJson(ErrorResponse(error.message)))
```

Comment:
In fact, I'm going to do this in a separate PR as there are probably more cases than just this.